contracts/SushiYieldSource.sol

```diff
@@ -1,29 +1,36 @@
 // SPDX-License-Identifier: GPL-3.0
 
 pragma solidity 0.6.12;
-
 import { IYieldSource } from "@pooltogether/yield-source-interface/contracts/IYieldSource.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
 import "@openzeppelin/contracts/math/SafeMath.sol";
 
 import "./ISushiBar.sol";
-import "./ISushi.sol";
 
 /// @title A pooltogether yield source for sushi token
 /// @author Steffel Fenix
 contract SushiYieldSource is IYieldSource {
-    
+    using SafeERC20 for IERC20;
     using SafeMath for uint256;
     
     ISushiBar public immutable sushiBar;
-    ISushi public immutable sushiAddr;
+    IERC20 public immutable sushiAddr;
     
     mapping(address => uint256) public balances;
 
-    constructor(ISushiBar _sushiBar, ISushi _sushiAddr) public {
+    constructor(ISushiBar _sushiBar, IERC20 _sushiAddr) public {
         sushiBar = _sushiBar;
         sushiAddr = _sushiAddr;
     }
 
+
+    ///@notice Approve SUSHI to spend infinite sushiBar (xSUSHI)
+    /// @dev No initializer flag required
+    function intialize() external {
+        sushiAddr.safeApprove(address(sushiBar), type(uint256).max);
```

Comment:
Is this safe or do we need the intiailizer modifier?

Reply:
@asselstine Is it safe to have public access to this function?

Reply:
Yes, because the SushiBar is an immutable contract that is very simple